src/block/utils/StructureVoidType.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block\utils;
+
+enum StructureVoidType{
+	case VOID;
+	case AIR;
```

Comment:
What's the difference between these two types?

Reply:
From the [minecraft wiki](https://minecraft.wiki/w/Structure_Air#Usage):
> It exists as a data value of [structure voids](https://minecraft.wiki/w/Structure_void). It acts similarly to structure voids. When holding a structure air or structure void, it becomes visible. Structure air and structure voids can stack together, the icon of the stack is the icon of whichever block was most recently added to the stack. Transparent blocks can be placed on them just like with structure voids. In the inventory, structure air is called structure void. If you give yourself a structure air then whenever you hold a structure void it will look like a structure air in your hand in both third and first person. Structure air are copied by structure blocks. This makes structure air very useful so always use structure air for your creative builds instead of structure voids. Note: structure air becomes structure voids when placed by hand, but if placed with /[setblock](https://minecraft.wiki/w/Commands/setblock) ~ ~ ~ structure_void ["structure_void_type":"air"] it will stay as a structure air.

I was not able to completely verify this information, the difference that matters in this pr is that vanilla minecraft will forbid you from placing structure airs, and in this pr you can hold them and place them fine.

Reply:
And you can't stack two different types in the pr

Reply:
The difference is that structure air will replace whatever block is at that position with an air block, while structure void will just leave it be

Reply:
That's exactly what normal air does ...

Reply:
Depends on the structure block settings iirc

Reply:
Meaning what?

Reply:
I was thinking about the Remove Blocks setting, but it doesn't help (its just another weird setting).

After testing on 1.20.41.02, I was unable to find a useful difference between structure air and structure void. I was able to reproduce normal behavior with both:
> Structure voids can be placed as part of a structure and are ignored when the structure file is loaded. This results in the blocks at the location where the structure is placed being maintained and not being overridden.

This is exactly what happened :

> For example, if a player saves a 2 block high structure with air at the top and a structure void at the bottom, and loads it into a 2 block high area of stone, the top block becomes air but the bottom stays stone.

What did **not** happen:
 - "Structure air are copied by structure blocks."

TL;DR I was not able to find any difference in the structure block usage. I left a comment on the wiki page

Reply:
Did you try pasting structures underwater with this? I wonder if there's a difference in the water displacement behaviour.